Vehicle/Vehicle.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace AutonomousCars
+{
+    public class Vehicle : IVehicle
+    {
+        private IVehicleInformation _vehicleInformation;
+
+        public Vehicle(IVehicleInformation vehicleInformation)
+        {
+            _vehicleInformation = vehicleInformation;
+        }
+
+        public IVehicleInformation VehicleInformation { get => _vehicleInformation; }
+
+        public double CalculateDistance(IVehicle vehicle)
+        {
+            if (vehicle == null || vehicle.VehicleInformation.GPSData == null)
+            {
+                throw new ArgumentException("Invalid vehicle!");
+            }
+
+            // Calculate Euclid's distance to vehicle
+            return Math.Sqrt(
+                Math.Pow(_vehicleInformation.GPSData.Latitude - vehicle.VehicleInformation.GPSData.Latitude, 2) + 
+                Math.Pow(_vehicleInformation.GPSData.Longitude - vehicle.VehicleInformation.GPSData.Longitude, 2) +
+                Math.Pow(_vehicleInformation.GPSData.Altitude - vehicle.VehicleInformation.GPSData.Altitude, 2)
+                );
+        }
+
+        public IVehicle GetClosestVehicle(ICollection<IVehicle> vehicles)
+        {
+            if (vehicles.Count == 0)
+            {
+                throw new ArgumentException("Empty vehicle collection!");
+            }
+
+            IVehicle closestVehicle = null;
+            double closestDistance = double.MaxValue;
+
+            foreach (IVehicle vehicle in vehicles)
+            {
+                if (vehicle == null)
+                {
+                    continue;
+                }
+
+                double distance = CalculateDistance(vehicle);
+                if (distance <= closestDistance)
+                {
+                    closestDistance = distance;
+                    closestVehicle = vehicle;
+                }
+            }
+
+            return closestVehicle;
+        }
+
+        public IVehicleInformation ReceiveInformation(IVehicle vehicle)
```

Comment:
This is not doing much with the information. I'd expect this kind of method to store the data for later use.

Interesting return type pick. Would be interested why choose the VehicleInformation and not the Vehicle / void types.